### PR TITLE
FTb/BL documentation notebook

### PR DESCRIPTION
**Pull request by mpharrigan**
This PR introduces a notebook that walks through the loschmidt echo benchmark end-to-end, and touches on some other key concepts.

### CODE REVIEW

File: docs/ftbbl.ipynb [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72765c1f",
+   "metadata": {},
+   "source": [
+    "### Unique `executable_family`\n",
+    "\n",
+    "Each benchmark has a globally-unique `executable_family` string, as well as other helpful identifiers. For example: the `recirq.otoc.loschmidt.tilted_square_lattice` has the domain of `recirq.otoc`, as it is similar to circuits run as part of the [OTOC Experiment](https://quantumai.google/cirq/experiments/otoc). We'll select this one for further inspection."
```
Comment (dstrain115) [inline docs/ftbbl.ipynb:52]: It feels like this is missing a sentence that explains what an `executable_family` string actually is (and perhaps its scope, like how broad or narrow is a family).  Or maybe move the below "In ReCirq, the executable_family is the same as the Python module path" up here instead?

File: docs/ftbbl.ipynb [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+    "plt.ylabel('Success Probability')\n",
+    "plt.tight_layout()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9b4b73a3",
+   "metadata": {},
+   "source": [
+    "## Appendix: `QuantumExecutable` and `ExecutableSpec`\n",
```
Comment (dstrain115) [inline docs/ftbbl.ipynb:403]: Even after reading this, I am still a little confused about this.  I think a diagram (as mentioned above) would really help.

**Review by dstrain115 — COMMENTED** [review]
This looks really good!  Exciting to see this coming together.  I left some comments about different things that could be improved in the tutorial.

File: docs/ftbbl.ipynb [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dfa69118",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import recirq.algorithmic_benchmark_library as algos\n",
+    "\n",
+    "for algo in algos.BENCHMARKS:\n",
+    "    display(algo)"
```
[the remark this answers is not in the document]
Comment (mpharrigan) [reply docs/ftbbl.ipynb:42]: This is a builtin in jupyter notebooks. I added an `import` statement to make it explicit

File: docs/ftbbl.ipynb (hunk 2 above)
Comment (mpharrigan) [reply docs/ftbbl.ipynb:469]: working on some diagrams for this, stay tuned.

**Review by dstrain115 — APPROVED** [review]
LGTM.  Cool diagrams!

**Pull request merged**